pkg/cloudprovider/provider/digitalocean/provider.go

```diff
@@ -268,19 +266,20 @@ func (p *provider) Create(machine *v1alpha1.Machine, userdata string) (instance.
 	ctx := context.TODO()
 	client := getClient(c.Token)
 
-	// TODO(p0lyn0mial): an ssh-key should be created only if the CoreOS was requested
-	// TODO(p0lyn0mial): an ssh-key must be temporal - deleted after an instance has been created.
-	if p.privateKey == nil {
-		var err error
-		p.privateKey, err = machinessh.NewPrivateKey("machine-controller")
-		if err != nil {
-			return nil, err
-		}
+	tmpSSHKeyPair, err := machinessh.NewPrivateKey("machine-controller-do-tmp-key")
+	if err != nil {
+		return nil, err
 	}
-	fingerprint, err := ensureSSHKeysExist(ctx, client.Keys, p.privateKey)
+	fingerprint, err := ensureSSHKeysExist(ctx, client.Keys, tmpSSHKeyPair)
```

Comment:
`ensureSSHKeysExist` can be deleted. It contains more code than actual necessary anymore.

I would put the `client.Keys.Create` in here and generate a random name.